backend-nest/apps/gateway/src/gateway.module.ts

```diff
@@ -1,24 +1,41 @@
 import { Module } from '@nestjs/common';
 
-import { GatewayController } from './gateway.controller';
-import { GatewayService } from './gateway.service';
-import { ClientsModule, Transport } from '@nestjs/microservices';
+import { ConfigModule, ConfigService } from '@nestjs/config';
 import { JwtModule } from '@nestjs/jwt';
+import { ClientsModule, Transport } from '@nestjs/microservices';
 import { PassportModule } from '@nestjs/passport';
-import { ConfigModule, ConfigService } from '@nestjs/config';
-import { JwtStrategy } from './strategies/jwt.strategy';
 import { AuthModule } from './auth/auth.module';
+import { GatewayController } from './gateway.controller';
+import { GatewayService } from './gateway.service';
+import { JwtStrategy } from './strategies/jwt.strategy';
 
 @Module({
   imports: [
-    ClientsModule.register([
-      { name: 'EMPLOYEES_SERVICE', transport: Transport.TCP },
-    ]),
-    PassportModule,
     ConfigModule.forRoot({
       envFilePath: './apps/gateway/.env',
       isGlobal: true,
     }),
+    ClientsModule.register([
+      {
+        name: 'EMPLOYEES_SERVICE',
+        transport: Transport.TCP,
+        options: {
+          port: parseInt(process.env.EMPLOYEES_SERVICE_PORT),
+        },
+      },
+      {
+        name: 'MAILER_SERVICE',
+        transport: Transport.RMQ,
+        options: {
+          urls: [process.env.RABBITMQ_URL],
+          queue: 'holiday-request',
```

Comment:
same goes here, might be better to extract those queue names into a constants somewhere, becuase if you need to change the value down the line its going to be a bit messy to change them everywhere in the code

Reply:
Should I make a common .env file which will be used by multiple services or that's not a good idea? Make it a common constant variable inside the common library instead?

Reply:
well imo this queue is not something which will have to change without a pr or other additional code changes, a .env would make sens if you want to change it fast without having to make a pr, and a new deployment.
If you think that its something along those lines you can use a .env otherwise a constant or readonly object, or an enum might be a better

Reply:
Well, I don't think either that it gonna change frequently, so leave in a constant for now.